Objects/abstract.c

```diff
@@ -2881,7 +2881,47 @@ PyAIter_Check(PyObject *obj)
             tp->tp_as_async->am_anext != &_PyObject_NextNotImplemented);
 }
 
+static int
+iternext(PyObject *iter, PyObject **item)
+{
+    *item = (*Py_TYPE(iter)->tp_iternext)(iter);
+    if (*item == NULL) {
```

Comment:
For readability (to remove one indentation level), maybe start with the common case:

```c
if (*item != NULL) {
    return 1;
}
```